src/server/handlers/caspar-cg-handler.ts

```diff
@@ -280,23 +280,29 @@ function startTimeEmitInterval() {
 
 async function startFileChangesPollingInterval(): Promise<void> {
     await pollFileChanges()
-    setInterval(() => {
+    await pollThumbnailChanges()
+    setInterval(async () => {
         if (!waitingForCcgResponse) {
             waitingForCcgResponse = true
-            pollFileChanges().then(() => {
-                waitingForCcgResponse = false
-            })
+            await pollFileChanges()
+            await pollThumbnailChanges()
+            waitingForCcgResponse = false
         }
     }, 3000)
 }
 
 async function pollFileChanges(): Promise<void> {
+    logger.trace('Polling for File Changes.')
+    await loadFileList()
+}
+
+async function pollThumbnailChanges(): Promise<void> {
+    logger.trace('Polling for Thumbnail Changes.')
     const newThumbnails = await loadCcgThumbnails()
     if (newThumbnails.hasChanges) {
         thumbnails = newThumbnails.thumbnails
         assignThumbnailsToOutputs()
     }
```

Comment:
Just a small optimization: This can be changed into a guard.
```suggestion
    if (!newThumbnails.hasChanges) {
        return
    }
    thumbnails = newThumbnails.thumbnails
    assignThumbnailsToOutputs()
```

Reply:
Done